packages/webapp/src/containers/revenueTypeSlice.js

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { createEntityAdapter, createSlice } from '@reduxjs/toolkit';
+import { loginSelector, onLoadingFail, onLoadingStart } from './userFarmSlice';
+import { pick } from '../util/pick';
+import { createSelector } from 'reselect';
+
+export const getRevenueType = (obj) => {
+  return pick(obj, [
+    'revenue_type_id',
+    'revenue_name',
+    'revenue_translation_key',
+    'farm_id',
+    'deleted',
+  ]);
+};
+
+const addManyRevenueTypes = (state, { payload: revenueTypes }) => {
+  state.loading = false;
+  state.error = null;
+  state.loaded = true;
+  revenueTypeAdapter.upsertMany(
```

Comment:
This copies the taskTypeSlice but for some reason I thought we figured that get requests should be setAll ?

Reply:
Yeah I guess I thought we figured that the way you found for task and managementPlan was the best way forward.

Reply:
I'm actually thinking of backtracking on both `task` and `managementPlan` in favour of upsert with soft delete, given the crashing selectors! I'll organize my thoughts on this and bring it up in tech daily for discussion.